### PR TITLE
Update Azure Batch enum to use "values" - also some documentation updates

### PR DESCRIPTION
**Pull request by matthchr**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by matthchr** [comment]
@xingwu1 @darylmsft - Please take a look 

@smith1511 - Take a look at the commit for NATPool docs

File: specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json [hunk 1]
```diff
@@ -10391,7 +10391,19 @@
           ],
           "x-ms-enum": {
             "name": "OSType",
-            "modelAsString": false
+            "modelAsString": false,
+            "values": [
+              {
+                "value": "linux",
+                "description": "The Linux operating system.",
+                "name": "linux"
```
Comment (xingwu1) [inline specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json]: Don't need specify "name" if not override the enum name.

File: specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json [hunk 2]
```diff
@@ -11453,14 +11537,26 @@
...
+          "description": "The default value is currentuser. This property is applicable only for pools configured with Windows nodes (that is, created with cloudServiceConfiguration, or with virtualMachineConfiguration using a Windows image reference). For Linux compute nodes, the certificates are stored in a directory inside the task working directory and an environment variable AZ_BATCH_CERTIFICATES_DIR is supplied to the task to query for this location. For certificates with visibility of 'remoteUser', a 'certs' directory is created in the user's home directory (e.g., /home/{user-name}/certs) and certificates are placed in that directory.",
```
Comment (xingwu1) [inline specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json]: currentUser vs currentuser in the "description"? None of it is good.

File: specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json [hunk 3]
```diff
@@ -12498,14 +12644,26 @@
         "validationStatus": {
           "type": "string",
           "title": "Whether the task counts have been validated.",
-          "description": "If the validationStatus is unvalidated, then the Batch service has not been able to check state counts against the task states as reported in the List Tasks API. The validationStatus may be unvalidated if the job contains more than 200,000 tasks.",
+          "description": "",
```
Comment (xingwu1) [inline specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json]: Why empty here?

File: specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json [hunk 4]
```diff
@@ -12664,7 +12822,7 @@
         },
         "storageAccountType": {
           "title": "The storage account type to be used for the data disk.",
-          "description": "If omitted, the default is \"Standard_LRS\". Values are:\n\n Standard_LRS - The data disk should use standard locally redundant storage.\n Premium_LRS - The data disk should use premium locally redundant storage.",
+          "description": "If omitted, the default is \"Standard_LRS\".",
```
Comment (xingwu1) [inline specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json]: inconsistent. Some places use all lower cases. Here is different.

**Comment by xingwu1** [comment]
Major issue: I tried this spec with different languages. C# looks good. However, neither java or python honor this extension. So we can approve this PR until major languages support it.

**Comment by darylmsft** [comment]
I think/hope Xing means "cannot approve..."  right?

File: specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json (hunk 1 above)
Comment (matthchr) [reply specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json]: I will fix this

**Comment by matthchr** [comment]
@xingwu1 I knew that Python didn't honor this -- can you file (or help me file) a GitHub issue for Java like the ones I filed for [node](https://github.com/Azure/autorest/issues/2706) and [python](https://github.com/Azure/autorest/issues/2704)?

I already have commitment from @amarzavery to fix Node.js this month, and I spoke with @annatisch who says that we should be able to have the fix for Python within relatively short order as well.

Given that this also fixes the C# SDK and is just generally more correct it seemed better to move and get the tooling fixed by the time we regenerate new SDKs (which probably wont be for a while).

File: specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json (hunk 3 above)
Comment (matthchr) [reply specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json]: Fixed

File: specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json (hunk 4 above)
Comment (matthchr) [reply specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json]: Fixed

File: specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json [hunk 5]
```diff
@@ -14804,51 +15198,115 @@
...
+        ]
       }
     },
     "OnTaskFailure": {
       "type": "string",
       "title": "The action the Batch service should take when any task in the job fails.",
-      "description": "A task is considered to have failed if has a failureInfo. A failureInfo is set if the task completes with a non-zero exit code after exhausting its retry count, or if there was an error starting the task, for example due to a resource file download error. Values are:\n\n noAction - do nothing.\n performExitOptionsJobAction - take the action associated with the task exit condition in the task's exitConditions collection. (This may still result in no action being taken, if that is what the task specifies.)\n\n The default is noAction.",
+      "description": "A task is considered to have failed if has a failureInfo. A failureInfo is set if the task completes with a non-zero exit code after exhausting its retry count, or if there was an error starting the task, for example due to a resource file download error. The default is noAction.",
       "enum": [
         "noAction",
```
Comment (mcardosos) [inline specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json]: >noAction [](start = 9, length = 8)

`noaction` :D

**Review by mcardosos — CHANGES_REQUESTED** [review]
Examples should also be [updated](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/302225069#L689)

**Comment by mcardosos** [comment]
@matthchr many of this changes have to do with documentation, but some other are because the swagger was not describing correctly the API, am I right?

File: specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json (hunk 5 above)
Comment (matthchr) [reply specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json]: Nice catch

**Comment by matthchr** [comment]
@mcardosos - Basically yes - the Swagger was using a camelCased name because we had to in order to get well generated code. But the service actually doesn't return the text camelCase, it returns it all lowercase. BUT in all the generated languages they were case insensitive here anyway so it didn't matter.

Newtonsoft 10 came out and changed that in C# (now sometimes it IS case sensitive), so we needed to update the spec to more accurately match the behavior of the service (while still keeping the nice naming we had before by using the `values` extension to `x-ms-enum`.)

**Comment by matthchr** [comment]
@mcardosos I updated the samples -- take a look again?

File: specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json [hunk 6]
```diff
@@ -14804,51 +15197,115 @@
...
       }
     },
     "OnAllTasksComplete": {
       "type": "string",
       "title": "The action the Batch service should take when all tasks in the job are in the completed state.",
-      "description": "Values are:\n\n noAction - do nothing. The job remains active unless terminated or disabled by some other means.\n terminateJob - terminate the job. The job's terminateReason is set to 'AllTasksComplete'.",
       "enum": [
-        "noAction",
+        "noaction",
         "terminateJob"
```
Comment (mcardosos) [inline specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json]: >terminateJob [](start = 9, length = 12)

terminatejob

File: specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json [hunk 7]
```diff
@@ -13933,29 +14154,95 @@
...
+              {
+                "value": "creating",
+                "description": "The Batch service has obtained the underlying virtual machine from Azure Compute, but it has not yet started to join the pool."
+              },
+              {
+                "value": "starting",
+                "description": "The Batch service is starting on the underlying virtual machine."
+              },
+              {
+                "value": "waitingForStartTask",
```
Comment (mcardosos) [inline specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json]: >waitingForStartTask [](start = 26, length = 19)

waitingforstarttask

File: specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json [hunk 8]
```diff
@@ -14606,15 +14940,30 @@
         "nodeDisableSchedulingOption": {
           "type": "string",
           "title": "What to do with currently running tasks when disabling task scheduling on the compute node.",
-          "description": "Values are:\n\n requeue - Terminate running task processes and requeue the tasks. The tasks may run again on other compute nodes, or when task scheduling is re-enabled on this node. Enter offline state as soon as tasks have been terminated.\n terminate - Terminate running tasks. The tasks will not run again. Enter offline state as soon as tasks have been terminated.\n taskcompletion - Allow currently running tasks to complete. Schedule no new tasks while waiting. Enter offline state when all tasks have completed.\n\n The default value is requeue.",
+          "description": "The default value is requeue.",
           "enum": [
             "requeue",
             "terminate",
             "taskCompletion"
```
Comment (mcardosos) [inline specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json]: >taskCompletion [](start = 13, length = 14)

taskcompletion

**Comment by mcardosos** [comment]
@matthchr Awesome
Thanks for updating the examples. There are just some tiny details with camel case and it should be good enough to merge :D

**Comment by matthchr** [comment]
Thanks @mcardosos for the good catches! I think things look okay now.

**Comment by mcardosos** [comment]
Great!
There is a [linter error](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/304869507#L732) in the swagger, but as it was not introduced on this PR, this PR is approved. Just mentioning this for awareness.

**Review by mcardosos — APPROVED** [review]

**Pull request merged**